util/src/main/java/io/kubernetes/client/util/authenticators/OpenIDConnectAuthenticator.java

```diff
@@ -113,13 +113,27 @@ public Map<String, Object> refresh(Map<String, Object> config) {
     String clientSecret = (String) config.getOrDefault(OIDC_CLIENT_SECRET, "");
     String idpCert = (String) config.get(OIDC_IDP_CERT_DATA);
 
+    
     SSLContext sslContext = null;
 
+    // Initialize a secure default SSLContext using the system default TrustManager.
+    // This ensures that we always have certificate validation even when no idp-certificate-authority-data is provided.
+    try {
```

Comment:
Shouldn't this new block be in the 'else' part of the following test, i.e.  if(idpCert != null) { existing block } else { this block }, otherwise this block runs even when there is an idpCert but the result is not used.   Unless there's some state outside of this method that changes based on this block and those changes are required for the following if(idpCert != null) block